src/state/router.js

```diff
@@ -0,0 +1,142 @@
+
+import { REQUEST_TRIGGER, API_VERSIONS_RECEIVE, REQUEST_SELECT_ENDPOINT } from './actions';
+import { defaultState as defaultRequestState } from './request/reducer';
+import { defaultState as defaultUiState } from './ui/reducer';
+import { loadEndpoints } from './endpoints/actions';
+import { getEndpoints } from './endpoints/selectors';
+
+function getUrlParams() {
+	return new URL( window.location.href ).searchParams;
+}
+
+function encodeString( val ) {
+	return ( val === null || val === undefined ) ? '' : encodeURIComponent( val );
+}
+
+function encodeObject( val ) {
+	return ( typeof val === 'object' ) && Object.keys( val ).length === 0 ? '' : encodeURIComponent( JSON.stringify( val ) );
+}
+
+function decodeString( val ) {
+	return decodeURIComponent( val );
+}
```

Comment:
is this offering much over directly calling `decodeURIComponent()`? I find that the abstraction hasn't made the code clearer to me, and from the way it's used, it seems like it's only being used in the context of a query string where `decodeURIComponent` is a fitting match.

Reply:
It seems that it's making a pair between `encodeString` and `decodeString`, it's clear that those do the inverse of each other. `encodeObject` and `decodeObject` is a complementary pair. Replacing one of those 4 with a direct call makes it more confusing imo.